glusterd2/commands/volumes/common.go

```diff
@@ -356,3 +358,38 @@ func isActionStepRequired(opt map[string]string, volinfo *volume.Volinfo) bool {
 
 	return false
 }
+
+func txnDeleteBrickVolfiles(c transaction.TxnCtx) error {
+	var volinfo volume.Volinfo
+	if err := c.Get("volinfo", &volinfo); err != nil {
+		c.Logger().WithError(err).WithField(
+			"key", "volinfo").Error("Failed to get key from store")
+		return err
+	}
+
+	err := volgen.DeleteBricksVolfiles(volinfo.GetLocalBricks())
+	if err != nil {
+		c.Logger().WithError(err).WithFields(log.Fields{
+			"template": "brick",
+			"volume":   volinfo.Name,
+		}).Error("failed to delete brick volfile")
```

Comment:
don't we need to return err here?

Reply:
Don't want fail transaction if this fails. Side effect is we may have extra volfiles on disk. Functionality is not affected.

Reply:
when the extra volfiles on disk will be deleted? because it may fill up the disk space in the long run

Reply:
long run and too many failures is unlikely. Anyways I will return error for now, if required we can skip in future.